lib/schema.js

```diff
@@ -540,6 +540,12 @@ Schema.prototype.add = function add(obj, prefix) {
         this.nested[prefix.substring(0, prefix.length - 1)] = true;
       }
       this.path(prefix + key, val);
+      if (Array.isArray(val) && val.length != 0 && val[0].discriminators) {
```

Comment:
There's a case we missed: `val` can be a Schema, and Schemas can have a `discriminators` property. Let's replace this with `if (utils.isPOJO(val[0]) && utils.isPOJO(val[0].discriminators))`